crates/astria-conductor/src/executor.rs

```diff
@@ -196,6 +213,18 @@ impl<C: ExecutionClient> Executor<C> {
             );
             return Ok(Some(execution_hash.clone()));
         }
+        // if the block that we just recieved is not the next block in the sequence to be executed,
+        if let Some(parent_block) = block.header.last_block_id {
```

Comment:
I think we need to track the parent hash of the `most_recently_executed_block_hash` as well, since we are allowed to execute a block that is the same number as the most recently executed block (in the case of a 1-block fork for example)

Reply:
maybe instead of storing just `most_recently_executed_block_hash` store the whole most recently executed sequencer header?

Reply:
Good point! My intention was to only keep track of the most recent safe block as the parent hash of all blocks that could be executed. I think i'm going to rename this and change this to reflect that